src/java.base/share/classes/java/util/concurrent/ForkJoinPool.java

```diff
@@ -2025,14 +2020,15 @@ else if (taken)
                                     else
                                         break;            // empty
                                 }
-                                Thread.onSpinWait();
+                                if (pb == (pb = b))       // base unchanged
+                                    Thread.onSpinWait();
```

Comment:
@DougLea Doesn't this mean that we'll always onSpinWait at least once here (since pb is initialized to -1 and not written until this check)

Reply:
No, almost the opposite: it only spinWaits if base is apparently stuck, except (harmlessly) every 4B encounters when it happens to be -1. (The value -1 works best here since base starts at 0 and is simpler than other ways of checking for stalls.)

Reply:
@DougLea You're right, I've seen too many of these consistency-checks to have my brain read "!=" automatically. Sorry for the noise :)